src/main/kotlin/org/rust/ide/inspections/lints/RsUnusedMustUseInspection.kt

```diff
@@ -0,0 +1,104 @@
+    override fun getFamilyName() = RsBundle.message("inspection.UnusedMustUse.FixAddLetUnderscore.name")
+
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+        val stmt = descriptor.psiElement as RsExprStmt
+        stmt.replace(RsPsiFactory(project).createLetDeclaration("_", stmt.expr))
+    }
+}
+
+private class FixAddUnwrap : LocalQuickFix {
+    override fun getFamilyName() = RsBundle.message("inspection.UnusedMustUse.FixAddUnwrap.name")
+
+    override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+        val stmt = descriptor.psiElement as RsExprStmt
+        stmt.expr.replace(RsPsiFactory(project).createExpression("${stmt.expr.text}.unwrap()"))
+    }
+}
+
+private class FixAddExpect(anchor: PsiElement) : LocalQuickFixAndIntentionActionOnPsiElement(anchor) {
+    override fun getFamilyName() = RsBundle.message("inspection.UnusedMustUse.FixAddExpect.family.name")
+    override fun getText() = familyName
+
+    override fun invoke(project: Project, file: PsiFile, editor: Editor?, startElement: PsiElement, endElement: PsiElement) {
+        val dotExpr = RsPsiFactory(project).createExpression("${startElement.text}.expect(\"\")")
+        val newDotExpr = startElement.replace(dotExpr) as RsDotExpr
+        val expectArgs = newDotExpr.methodCall?.valueArgumentList?.exprList
+        val stringLiteral = expectArgs?.singleOrNull() as RsLitExpr
+        val template = editor?.newTemplateBuilder(newDotExpr) ?: return
+        val rangeWithoutQuotes = TextRange(1, stringLiteral.textRange.length - 1)
+        template.replaceElement(stringLiteral, rangeWithoutQuotes, "TODO: panic message")
+        template.runInline()
+    }
+}
+
+private class InspectionResult(val description: String, val fixes: List<LocalQuickFix>)
+
+private fun inspectAndProposeFixes(expr: RsExpr): InspectionResult? {
+    val mustUseAttrName = "must_use"
+    val type = expr.type as? TyAdt
+    val func = when (expr) {
+        is RsDotExpr -> expr.methodCall?.getFunctionCallContext()?.function
+        is RsCallExpr -> expr.getFunctionCallContext()?.function
+        else -> null
+    }
+    val attrType = type?.item?.findFirstMetaItem(mustUseAttrName)
+    val attrFunc = func?.findFirstMetaItem(mustUseAttrName)
+    val description = when {
+        attrType != null -> RsBundle.message("inspection.UnusedMustUse.description.type.attribute", type)
+        attrFunc != null -> RsBundle.message("inspection.UnusedMustUse.description.function.attribute", func.name.toString())
+        else -> return null
+    }
+    val fixes: MutableList<LocalQuickFix> = mutableListOf(FixAddLetUnderscore())
+    if (expr.returnsStdResult()) {
+        fixes += FixAddExpect(expr)
+        fixes += FixAddUnwrap()
+    }
+    return InspectionResult(description, fixes)
+}
+
+/** Analogue of rustc's unused_must_use. See also [RsDoubleMustUseInspection]. */
+class RsUnusedMustUseInspection : RsLintInspection() {
```

Comment:
I wonder if a weak warning would be a better highlighting level for this inspection?

Reply:
Done!